### PR TITLE
refresh and fetch description and profile image

### PR DESCRIPTION
**Pull request by benny-conn**
Changes:

- **Changed** contract dedupe to check for all metadata fields across all contracts received
- **Added** collection image url support to opensea and objkt providers
- **Added** script for refreshing contracts

### CODE REVIEW

File: service/multichain/multichain.go [hunk 1]
```diff
@@ -1539,35 +1540,53 @@ func tokensToNewDedupedTokens(tokens []chainTokens, contracts []persist.Contract
...
+		res = append(res, persist.ContractGallery{
+			Chain:        address.Chain(),
+			Address:      address.Address(),
+			Symbol:       persist.NullString(meta.Symbol),
+			Name:         persist.NullString(meta.Name),
+			OwnerAddress: meta.OwnerAddress,
+			Description:  persist.NullString(meta.Description),
+		})
 	}
 	return res
```
Comment (benny-conn) [inline service/multichain/multichain.go:1591]: This is the most major change in this PR

**Review by Robinnnnn — COMMENTED** [review]
this is fantastic, benny! super pumped!

@radazen any concern with writing the `CreatorAddress` on the contract? guessing no because this is within a separate `Descriptors` field

**Comment by radazen** [comment]
> this is fantastic, benny! super pumped!
> 
> @radazen any concern with writing the `CreatorAddress` on the contract? guessing no because this is within a separate `Descriptors` field

Shouldn't be a problem! We already map the contract's `CreatorAddress` to the database's `owner_address` field; this just does a better job processing that data before we write it.

**Comment by benny-conn** [comment]
Closing remark for this PR, we should probably run the profile images through the media pipeline at some point instead of using opensea's profile images directly.

**Pull request merged**